File: TreasurersApp/Models/TransactionTypeActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class TransactionTypeActionResult : WebResult<TransactionType>
    {
        public TransactionTypeActionResult(bool success, List<string> statusMessages, TransactionType transactionType)
            :base(success, statusMessages, transactionType)
        {
        }
    }
}

File: TreasurersApp/ClientApp/src/app/models/TransactionTypeActionResult.ts
import { TransactionType } from './TransactionType';
import { TransactionTypeService } from '../maintenance/transaction-type-maintenance/transaction-type.service';

export class TransactionTypeActionResult {
  success = false;
  statusMessages: string[] = [];
  transactionType: TransactionType = null;

  constructor(private transactionTypeService: TransactionTypeService) {
    this.success = false;
    this.transactionType = this.transactionTypeService.newTransactionType();
    this.statusMessages = [];
  }
}
